io/concatenateHDF5.m

```diff
@@ -0,0 +1,157 @@
+        %%% 'downsmpFactor' argument
+        imageStack = uint16(zeros(floor(rows*downsmpFactor),floor(cols*downsmpFactor),oriFrames-trim(1)-trim(2)));
+        if(downsmpFactor == 1)
+            for j=1:oriFrames-trim(1)-trim(2)
+                tifFile.setDirectory(j+trim(1));
+                imageStack(:,:,j) = uint16(tifFile.read());
+            end
+        else
+            for j=1:oriFrames-trim(1)-trim(2)
+                tifFile.setDirectory(j+trim(1));
+                imageStack(:,:,j) = uint16(imresize(tifFile.read(),downsmpFactor,'bilinear'));
+            end
+        end
+        
+        %%% Replace any dropped frames with the frame immediately
+        %%% preceeding it
+        if(droppedFrames ~=0)
+            for j=1:length(droppedFrames)
+                disp('DroppedFrame');
+                droppedFrame = droppedFrames(j);
+                if((droppedFrame > trim(1)) && (droppedFrame < oriFrames-trim(2)))
+                    droppedFrame = droppedFrame+trim(1);
+                    frontStack = imageStack(:,:,1:droppedFrame-1);
+                    frontStack = cat(3,frontStack,imageStack(:,:,droppedFrame-1));
+                    backStack = imageStack(:,:,droppedFrame:end);
+                    newImageStack = cat(3,frontStack,backStack);
+                    clear imageStack
+                    imageStack = newImageStack;
+                end
+            end
+        end
+        
+        numFrames = size(imageStack,3);
+        [dRows,dCols] = size(imageStack(:,:,1));
+        
+        h5write(fullfile(outputDir,hdf5Name),dataset,imageStack,[1,1,totalFrames],[dRows,dCols,numFrames]);
+        
+        %%%%%%%%%%%%%% NEED TO FIX %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+        %%% Currently does not take into account added frames if some were
+        %%% originally dropped - only tested on cases without dropped
+        %%% frames
+        gateUpFrame = totalFrames + (frame_indices(trialCount,2) - frame_indices(trialCount,1) - trim(1));
+        gateDownFrame = totalFrames + (frame_indices(trialCount,3) - frame_indices(trialCount,1) - trim(1));
+        fprintf(outputFile,'%s %s %s %f %d %d %d %d\n',location_info{trialCount,1},...
+            location_info{trialCount,2},location_info{trialCount,3},time(trialCount),totalFrames,gateUpFrame,gateDownFrame,totalFrames+numFrames-1);
+        %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+        
+        fprintf('File %d of %d stored\n',i,size(list,1));
+        %%% Total frame count stored in hdf5 file
+        totalFrames = totalFrames+numFrames;
+        
+        %%% Total frame count corresponding to behavior text file
+        testFrames = testFrames+oriFrames;
+        
+        clear imageStack tifName tifInfo tifFile droppedFrames newImageStack
+    end
+end
+
+%%% Remove the extra frame count needed to index the hdf5 file
+totalFrames = totalFrames-1;
+
+h5create(fullfile(outputDir,hdf5Name),'/Parameters/Number of Frames',1);

```

Comment:
At this point, I do not think the matlab function allows non numeric data to be stored in an hdf5. Please let me know if you have found something different? Would you want to create a numeric mapping for the locations?
1 = north
2 = south
3 = east
4 = west
or something like that? If not, I could include a parameter saying if the trial was correct and not include the actual locations (although it seem that they will be needed later in the analysis).